benchmarks/search/radius_search.cpp

```diff
@@ -7,56 +9,161 @@
+    state.PauseTiming();
+    int searchIdx = radiusSearchIdx++ % cloudIn->size();
+    state.ResumeTiming();
+    kdtree.radiusSearch(
+        searchIdx, searchRadius, k_indices, k_sqr_distances, neighborLimit);
+  }
+}
+
+static void
+BM_KdTreeAll(benchmark::State& state,
+             const pcl::PointCloud<pcl::PointXYZ>::Ptr cloudIn,
+             const double searchRadius,
+             const size_t neighborLimit)
+{
+  pcl::search::KdTree<pcl::PointXYZ> kdtree;
+  kdtree.setInputCloud(cloudIn);
+
+  // Leaving indices empty to have it search through all points
+  pcl::Indices indices;
+  std::vector<pcl::Indices> k_indices;
+  std::vector<std::vector<float>> k_sqr_distances;
+  for (auto _ : state) {
     auto start_time = std::chrono::high_resolution_clock::now();
-    organizedNeighborSearch.radiusSearch(
-        (*cloudIn)[searchIdx], searchRadius, k_indices, k_sqr_distances);
+    kdtree.radiusSearch(
+        *cloudIn, indices, searchRadius, k_indices, k_sqr_distances, neighborLimit);
     auto end_time = std::chrono::high_resolution_clock::now();
-
-    radiusSearchTime +=
-        std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time)
-            .count();
+    std::chrono::duration<double> elapsed = end_time - start_time;
+    state.SetIterationTime(elapsed.count() / cloudIn->size());
   }
 
-  state.SetItemsProcessed(state.iterations());
-  state.SetIterationTime(
-      radiusSearchTime /
-      (state.iterations() * indices.size())); // Normalize by total points processed
+  state.SetItemsProcessed(cloudIn->size());
+}
+
+#if PCL_HAS_NANOFLANN
+static void
+BM_KdTreeNanoflann(benchmark::State& state,
+                   const pcl::PointCloud<pcl::PointXYZ>::Ptr cloudIn,
+                   const double searchRadius,
+                   const size_t neighborLimit)
+{
+  pcl::search::KdTreeNanoflann<pcl::PointXYZ> kdtree;
+  kdtree.setInputCloud(cloudIn);
+
+  int radiusSearchIdx = 0;
+
+  pcl::Indices k_indices;
+  std::vector<float> k_sqr_distances;
+  for (auto _ : state) {
+    state.PauseTiming();
+    int searchIdx = radiusSearchIdx++ % cloudIn->size();
+    state.ResumeTiming();
+    kdtree.radiusSearch(
+        searchIdx, searchRadius, k_indices, k_sqr_distances, neighborLimit);
+  }
 }
+#endif
 
 int
 main(int argc, char** argv)
 {
   if (argc < 2) {
     std::cerr << "No test file given. Please provide a PCD file for the benchmark."
               << std::endl;
+    print_help();
     return -1;
   }
 
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloudIn(new pcl::PointCloud<pcl::PointXYZ>);
+  pcl::PCDReader reader;
+  reader.read(argv[1], *cloudIn);
+
+  double searchRadius = 0.1;
+  if (argc > 2) {
+    try {
+      searchRadius = std::abs(std::stod(argv[2]));
+    } catch (const std::invalid_argument&) {
+      std::cerr << "Error: Invalid search radius. Setting it to a default of 0.1."
+                << std::endl;
+    }
+  }
+
+  size_t neighborLimit = std::numeric_limits<int>::max();
```

Comment:
```suggestion
  size_t neighborLimit = 0;
```
When there is no limit on the number of neighbors in radius search, the parameter should be set to 0 (zero is also the default value of the parameter, that is, if the user did not specify a value). With `std::numeric_limits<int>::max()`, KdTreeNanoflann currently seems to select a search method that takes much longer than necessary (which explains the 343 us in your results).